core/src/backend/audio/mixer.rs

```diff
@@ -7,6 +7,39 @@ use std::io::Cursor;
 use std::sync::{Arc, Mutex, RwLock};
 use swf::AudioCompression;
 
+/// Holds the last 2048 output audio frames. Frames can be written to it one by
+/// one, and the last completely filled 1024-wide window can be read from it.
```

Comment:
What was the motivating factor for mixing 1024 samples at a time? Does Flash Player specifically do this or does AS3 code depend on this behavior?

Reply:
I tested this in the following way:

I created an audio file with a repeating 1024-wide section: 512 "high" values (something like half the max amplitude), followed by 512 zero samples. After 20 repetitions of this section, I inserted an extra zero sample. This sound is played in a loop.
This resulted in the captured sample windows to be completely static for a short time (until the 20 repetitions run out), then all of them ticking by one "to the right", when the extra zero is encountered.
This told me that the capture windows are not in random points in time, but taken from successive fixed length buffers, which some integer multiple of 1024 in length.

I also tested the same with 2048-long sections (1024 high + 1024 zero), but then the captured buffer was not static - it was oscillating between two different sets of samples.
This told me that the captured buffer is shorter than 2048. So only 1024 was left as a possible option.

I also know this, because as the first ("high") half 1024-long section slides "out out view" (so only zeroes are captured in between two "ticks"), then the high values immediately start rolling back in from the left.

Now, it's possible that this depends somewhat on the audio device or OS that FP runs on, but this is at least how it behaves on my (Linux) system.

Reply:
That sounds plausible enough.